File: week5/python/SumOfLeftLeaves_404.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        
        self.left_sum = 0
        def traverse(node: TreeNode, isLeft: bool):
            if not node:
                return
            
            if not node.left and not node.right:
                if isLeft:
                    self.left_sum += node.val
                return
            
            traverse(node.left, True)
            traverse(node.right, False)
        
        traverse(root, False)
        return self.left_sum

File: week5/java/SumOfLeftLeaves_404.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        return sumLeft(root, false);
    }
    
    private int sumLeft(TreeNode root, boolean isLeft) {
        if (root == null)
            return 0;
        
        if (root.left == null && root.right == null) {
            if (isLeft)
                return root.val;
            return 0;
        } 
        return sumLeft(root.left, true) + sumLeft(root.right, false);
    }
}
